server/payment/client.go

```diff
@@ -54,6 +50,32 @@ func (c *Client) Pay(price int, desc string, succURL string, cancelURL string) (
 	return resp.ID, resp.InvoiceURL, nil
 }
 
+func (c *Client) GetStatus(url string) (string, error) {
+	httpReq, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return "", err
+	}
+	httpReq.Header.Set("x-api-key", c.apikey)
+	res, err := c.httpClient.Do(httpReq)
```

Comment:
Is there convenient lib we can use here for request timeout and retry?

Reply:
This is good comment actually.
What about we add some metrics collect latency and succ rate then we know what value to use?
I have concern that adding timeout or retry may make things worse. I am under the impression that the requests take a quite long duration(about 10s).

Reply:
I also added a 60 seconds timeout to the client.